src/image/loading_displaying.js

```diff
@@ -158,6 +159,341 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
+  // validate parameters
+  p5._validateParameters('saveGif', arguments);
+  // get the project's framerate
+  // if it is undefined or some non useful value, assume it's 60
+  let _frameRate = this._targetFrameRate;
+  if (_frameRate === Infinity || _frameRate === undefined || _frameRate === 0) {
+    _frameRate = 60;
+  }
+
+  // calculate delay based on frameRate
+  let gifFrameDelay = 1 / _frameRate * 1000;
+
+  // constrain it to be always greater than 20,
+  // otherwise it won't work in some browsers and systems
+  // reference: https://stackoverflow.com/questions/64473278/gif-frame-duration-seems-slower-than-expected
+  gifFrameDelay = gifFrameDelay < 20 ? 20 : gifFrameDelay;
+
+  // because the input was in seconds, we now calculate
+  // how many frames those seconds translate to
+  let nFrames = Math.ceil(seconds * _frameRate);
+  let nFramesDelay = Math.ceil(delay * _frameRate);
+
+  //   initialize variables for the frames processing
+  var count = nFramesDelay;
+
+  this.noLoop();
+  // we start on the frame set by the delay argument
+  frameCount = nFramesDelay;
+
+  const lastPixelDensity = this._pixelDensity;
+  this.pixelDensity(1);
+
+  // We first take every frame that we are going to use for the animation
+  let frames = [];
+
+  if (document.getElementById('progressBar') !== null)
+    document.getElementById('progressBar').remove();
+
+  let p = this.createP('');
+  p.id('progressBar');
+
+  p.style('font-size', '16px');
+  p.style('font-family', 'Montserrat');
+  p.style('background-color', '#ffffffa0');
+  p.style('padding', '8px');
+  p.style('border-radius', '10px');
+  p.position(0, 0);
+
+  let pixels;
+  let gl;
+  if (this.drawingContext instanceof WebGLRenderingContext) {
+    // if we have a WEBGL context, initialize the pixels array
+    // and the gl context to use them inside the loop
+    gl = document.getElementById('defaultCanvas0').getContext('webgl');
+    pixels = new Uint8Array(gl.drawingBufferWidth * gl.drawingBufferHeight * 4);
+  }
+
+  while (count < nFrames + nFramesDelay) {
+    /*
+      we draw the next frame. this is important, since
+      busy sketches or low end devices might take longer
+      to render some frames. So we just wait for the frame
+      to be drawn and immediately save it to a buffer and continue
+      */
+    this.redraw();
+
+    // depending on the context we'll extract the pixels one way
+    // or another
+    let data = undefined;
+
+    if (this.drawingContext instanceof WebGLRenderingContext) {
+      pixels = new Uint8Array(
+        gl.drawingBufferWidth * gl.drawingBufferHeight * 4
+      );
+      gl.readPixels(
+        0,
+        0,
+        gl.drawingBufferWidth,
+        gl.drawingBufferHeight,
+        gl.RGBA,
+        gl.UNSIGNED_BYTE,
+        pixels
+      );
+
+      data = _flipPixels(pixels);
+    } else {
+      data = this.drawingContext.getImageData(0, 0, this.width, this.height)
```

Comment:
I think there's a slight performance cost to getting and storing the pixels array each frame. This is fine for sketches that use `frameCount`, but for ones that use `millis()` (so they drop frames to keep timing correct rather than letting timing slow down to not drop frames), do you think it's worth changing how `millis()` works while saving a gif? We could use fake time increments so the sketch thinks only `1000/targetFrameRate` milliseconds pass each frame to ensure the resulting gif doesn't look sped up because it took slightly longer than usual to render each frame. (Maybe this could be something for a follow-up PR since on its own this feature is super useful)

Reply:
I had no idea `millis()` worked like that! As for your question, I am not sure what to do about it.

Another related thing is that the delay functionality implemented will only work with sketches that use `frameCount`, as you said. I had a sketch from a friend that just updated the speed from some particles by adding up a value each frame and, since the simulation was not depending on the `frameCount` value, it just does nothing. 

Although it doesn't break, it does not behave as expected but have no idea how to solve it. :/

Reply:
Oh interesting, do you have a link to that sketch or an example that works similarly?

Also for `millis`, I think a solution would be, when we start recording, to replace the implementation of `millis` with a fake implementation. something like:
```js
// Replace with a fake implementation
const oldMillis = p5.prototype.millis
let time = millis()
p5.prototype.millis = () => time

while (count < nFrames + nFramesDelay) {
  time += 1000 / this._targetFrameRate // Fake time increment
  // render the frame here
}

// Put the old function back
p5.prototype.millis = oldMillis
```